File: flutter-sdk/lib/src/ui/chat/view_models/audio/audio_event.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

import 'package:chat_flutter_sdk/src/domain/models/chat_message/chat_message.dart';
import 'package:equatable/equatable.dart';

sealed class AudioEvent {}

final class AudioAmplitudeSubscribe extends AudioEvent with EquatableMixin {
  @override
  List<Object?> get props => [];
}

// Event that starts recording audio
final class AudioStartRecording extends AudioEvent with EquatableMixin {
  @override
  List<Object?> get props => [];
}

// Event that stops recording audio
final class AudioStopRecording extends AudioEvent with EquatableMixin {
  @override
  List<Object?> get props => [];
}

// Event that starts playing an audio
final class AudioPlay extends AudioEvent with EquatableMixin {
  final ChatMessage message;
  AudioPlay({required this.message});
  @override
  List<Object?> get props => [message];
}

// Event that stops playing audio
final class AudioStop extends AudioEvent with EquatableMixin {
  @override
  List<Object?> get props => [];
}

// Event to subscribe to a stream that emits events if an audio has been stopped playing
final class AudioCompletedSubscribe extends AudioEvent with EquatableMixin {
  @override
  List<Object?> get props => [];
}

File: android-sdk/sdk/src/main/kotlin/com/yalo/chat/sdk/ui/chat/AudioEvent.kt
// Copyright (c) Yalochat, Inc. All rights reserved.

package com.yalo.chat.sdk.ui.chat

import com.yalo.chat.sdk.domain.model.ChatMessage

// Port of flutter-sdk/lib/src/ui/chat/view_models/audio/audio_event.dart
sealed class AudioEvent {
    // Subscribe to the playback-completion stream (called once on screen entry).
    data object SubscribeToPlaybackCompletion : AudioEvent()
    // Start recording a voice message.
    data object StartRecording : AudioEvent()
    // Stop recording and return the resulting AudioData via state.
    data object StopRecording : AudioEvent()
    // Play the voice message; stops any currently playing message first.
    data class Play(val message: ChatMessage) : AudioEvent()
    // Stop / pause the currently playing message.
    data object Stop : AudioEvent()
}
